### PR TITLE
doc: Add issue templates for bug and feature request

### LINKED ISSUE #16627: doc: Move to new GitHub issue templates
Apparently we're using the old version of the GitHub issue templates. Not sure if there are any advantages to switching to the new form or not. Info here: https://help.github.com/en/articles/about-issue-and-pull-request-templates.

Screen Shot 2019-08-16 at 10 21 23 am

### PR DESCRIPTION
**Pull request by maflcko**
Fixes #16627 

Can be tested via https://github.com/MarcoFalke/bitcoin/issues

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK, I like how this provides the help text in comments instead of in-line, making it harder to have it accidentally end up in the issue text itself.

**Comment by promag** [comment]
Concept ACK.

**Comment by emilengler** [comment]
Concept ACK

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+name: Bug report
+about: Create a report to help us improve
+title: ''
+labels: bug
```
Comment (ch4ot1c) [inline .github/ISSUE_TEMPLATE/bug_report.md]: `Bug` in https://github.com/bitcoin/bitcoin/labels

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+<!--- How reliably can you reproduce the issue, what are the steps to do so? -->
+
+**System information**
+
+<!-- What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)? -->
+
+<!-- What type of machine are you observing the error on (OS/CPU and disk type)? -->
+
+<!-- For the GUI-related issue on Linux provide names and versions of a distro, a desktop environment and a graphical shell (if relevant). -->
```
Comment (ch4ot1c) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```
GUI-related issue? What is your operation system and its version? If Linux, what is your desktop environment and graphical shell?
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+<!-- Any extra information that might be useful in the debugging process. -->
```
Comment (ch4ot1c) [inline .github/ISSUE_TEMPLATE/bug_report.md]: `.` -> `?`

**Review by ch4ot1c — COMMENTED** [review]
ACK; Some suggestions

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (MarcoFalke) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Good catch

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (MarcoFalke) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Changed

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (MarcoFalke) [reply .github/ISSUE_TEMPLATE/bug_report.md]: I think it is fine as is

**Comment by MarcoFalke** [comment]
Addressed feedback by @ch4ot1c

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ If the node is "stuck" during sync or giving "block checksum mismatch" errors, p
 
 <!-- What type of machine are you observing the error on (OS/CPU and disk type)? -->
 
-<!-- For the GUI-related issue on Linux provide names and versions of a distro, a desktop environment and a graphical shell (if relevant). -->
+<!-- GUI-related issue? What is your operation system and its version? If Linux, what is your desktop environment and graphical shell? -->
 
 <!-- Any extra information that might be useful in the debugging process. -->
 <!--- This is normally the contents of a `debug.log` or `config.log` file. Raw text or a link to a pastebin type site are preferred. -->
```
Comment (ch4ot1c) [inline .github/ISSUE_TEMPLATE.md]: These `<!--- ` could be explained as a leading MD `- `, but don't look great in the issue preview plaintext and never get rendered as Markdown. Maybe instead add a `:` e.g. `Describe the issue:`?

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -17,7 +17,7 @@ If the node is "stuck" during sync or giving "block checksum mismatch" errors, p
 
 <!-- What type of machine are you observing the error on (OS/CPU and disk type)? -->
 
-<!-- For the GUI-related issue on Linux provide names and versions of a distro, a desktop environment and a graphical shell (if relevant). -->
+<!-- GUI-related issue? What is your operation system and its version? If Linux, what is your desktop environment and graphical shell? -->
```
Comment (ch4ot1c) [inline .github/ISSUE_TEMPLATE.md]: `operating`

**Review by ch4ot1c — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE.md (hunk 4 above)
Comment (MarcoFalke) [reply .github/ISSUE_TEMPLATE.md]: I think it is fine to keep this as it was previously

File: .github/ISSUE_TEMPLATE.md (hunk 5 above)
Comment (MarcoFalke) [reply .github/ISSUE_TEMPLATE.md]: Fixed typo

**Review by fanquake — APPROVED** [review]
ACK fabca7756d6908ad581f3a699f1be6ecc9f62e03

**Comment by jb55** [comment]
ACK fabca7756d6908ad581f3a699f1be6ecc9f62e03

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+**Expected behavior**
+
+<!--- What behavior did you expect? -->
+
+**Actual behavior**
+
+<!--- What was the actual behavior (provide screenshots if the issue is GUI-related)? -->
+
+**To reproduce**
```
Comment (jonatack) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Suggest: s/To reproduce/How to reproduce/

**Review by jonatack — COMMENTED** [review]
Concept ACK

**Comment by hebasto** [comment]
Concept ACK.
Could the legacy `.github/ISSUE_TEMPLATE.md` be removed at all?

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 6 above)
Comment (MarcoFalke) [reply .github/ISSUE_TEMPLATE/bug_report.md]: This is take from the "official" GitHub template, so I am leaving it for now (unless there is other stuff to fix up)

**Comment by MarcoFalke** [comment]
>  Could the legacy .github/ISSUE_TEMPLATE.md be removed at all?

Why? It doesn't cause any harm and it mentions useful advice for non-bug-reports (e.g. that support requests should go to stackexchange)

**Pull request merged**